dot/state/epoch.go

```diff
@@ -151,6 +172,91 @@ func NewEpochState(db database.Database, blockState *BlockState,
 	}, nil
 }
 
+// getNextEpochAndConfigDataFromDisk retrieves the next epoch and config data maps from the database
+func getNextEpochAndConfigDataFromDisk(db database.Database) (
```

Comment:
This function has duplicate code for `nextEpochData` and for `nextEpochConfig` can we refactor this func or create a new generic func that retrieves this data?

Reply:
The core of the function is 

```go
func restoreMapFromDisk[T types.NextEpochData | types.NextConfigDataV1](db (db database.Database, keyPrefix []byte) (resMap nextEpochMap[T], err error) {
	resMap := make(nextEpochMap[T])
	iter, err := db.NewPrefixIterator(keyPrefix)
	if err != nil {
		return resMap, err
	}
	defer iter.Release()

	for iter.First(); iter.Valid(); iter.Next() {
		next := new(T)
		next, epoch, fork, err := getNextEpochOrConfigData(next, iter)
		if err != nil {
			return resMap, err
		}

		if _, ok := nextConfigData[epoch]; !ok {
			nextConfigData[epoch] = make(map[common.Hash]T)
		}
		// Add data to the map
		resMap[epoch][fork] = *next
	}
}
```

And you can call it like:

```go
fn getNextEpochAndConfigDataFromDisk(db database.Database) (nextEpochMap[types.NextEpochData], nextEpochMap[types.NextConfigDataV1], error) {
    nextEpochDataMap, err := restoreMapFromDisk[types.NextEpochData](db, nextEpochDataPrefix)
     if err != nil {...}
     nextConfigDataMap, err := restoreMapFromDisk[types.NextConfigDataV1](db, nextConfigDataPrefix)
     if err != nil {...}
     return nextEpochDataMap, nextConfigDataMap, nil
}
```

Reply:
Solved!